src/peep/peep.c

```diff
@@ -10389,46 +10410,73 @@ void peep_update_name_sort(rct_peep *peep)
 	openrct2_reset_object_tween_locations();
 }
 
+void peep_sort()
+{
+	// Count number of peeps
+	uint16 sprite_index, num_peeps = 0;
+	rct_peep *peep;
+	FOR_ALL_PEEPS(sprite_index, peep) {
+		num_peeps++;
+	}
+
+	// No need to sort
+	if (num_peeps < 2)
+		return;
+
+	// Create a copy of the peep list and sort it using peep_compare
+	uint16 *peep_list = (uint16*)malloc(num_peeps * sizeof(uint16));
+	int i = 0;
+	FOR_ALL_PEEPS(sprite_index, peep) {
+		peep_list[i++] = peep->sprite_index;
+	}
+	qsort(peep_list, num_peeps, sizeof(uint16), peep_compare);
+
+	// Set the correct peep->next and peep->previous using the sorted list
+	for (i = 0; i < num_peeps; i++) {
+		peep = GET_PEEP(peep_list[i]);
+		peep->previous = (i > 0) ? peep_list[i - 1] : SPRITE_INDEX_NULL;
+		peep->next = (i + 1 < num_peeps) ? peep_list[i + 1] : SPRITE_INDEX_NULL;
+	}
+	// Make sure the first peep is set
+	gSpriteListHead[SPRITE_LIST_PEEP] = peep_list[0];
+
+	free(peep_list);
+
+	i = 0;
+	FOR_ALL_PEEPS(sprite_index, peep) {
+		i++;
+	}
+	assert(i == num_peeps);
```

Comment:
Can do, but it's only done once when loading a park. In  case there is anything wrong, we'll find out much sooner.